internal/app/machined/pkg/controllers/network/operator/dhcp4.go

```diff
@@ -93,10 +152,37 @@ func (d *DHCP4) Run(ctx context.Context, notifyCh chan<- struct{}) {
 			renewInterval = minRenewDuration
 		}
 
-		select {
-		case <-ctx.Done():
-			return
-		case <-time.After(renewInterval):
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(renewInterval):
+			case <-hostnameWatchCh:
+				err := d.updateHostname(ctx, &hostname)
```

Comment:
I've now removed `updateHostname` altogether and use the hostname directly from the event. Thanks for the pointer. The check needs to cover all event types, since hostname deletion should also indicate to the DHCP server that the node has lost its previous hostname. I'm doing a checked cast to `network.HostnameStatus`, which fails intentionally if it gets fed a `Tombstone` generated by a deletion event. I've also added a comment about this.